pilot/pkg/xds/ads.go

```diff
@@ -185,14 +176,14 @@ func (s *DiscoveryServer) receive(con *Connection, reqChannel chan *discovery.Di
 // handles 'push' requests and close - the code will eventually call the 'push' code, and it needs more mutex
 // protection. Original code avoided the mutexes by doing both 'push' and 'process requests' in same thread.
 func (s *DiscoveryServer) processRequest(req *discovery.DiscoveryRequest, con *Connection) error {
-
-	if !s.preProcessRequest(con.proxy, req) {
+	if req.TypeUrl == v3.HealthInfoType && features.WorkloadEntryHealthChecks {
```

Comment:
I think my comment got hidden in rebase, but I think the old code here was better, as it abstracts the details of health check

Reply:
@howardjohn Reverted